daemon/containerd/image_list_test.go

```diff
@@ -173,6 +173,33 @@ func TestImageListCheckTotalSize(t *testing.T) {
 	// TODO: This should also include the Size.Unpacked, but the test snapshotter doesn't do anything yet
 	assert.Check(t, is.Equal(all[0].Manifests[0].Size.Total, amd64ManifestSize+amd64ConfigSize+amd64LayerSize))
 	assert.Check(t, is.Equal(all[0].Manifests[1].Size.Total, amd64ManifestSize+amd64ConfigSize+amd64LayerSize))
+
+	t.Run("without layers", func(t *testing.T) {
+		var layers []ocispec.Descriptor
+		err = service.walkPresentChildren(ctx, img.Target, func(ctx context.Context, desc ocispec.Descriptor) error {
+			if c8dimages.IsLayerType(desc.MediaType) {
+				layers = append(layers, desc)
+			}
+			return nil
+		})
+		assert.NilError(t, err)
+
+		for _, layer := range layers {
+			err := cs.Delete(ctx, layer.Digest)
+			assert.NilError(t, err, "failed to delete layer %s", layer.Digest)
+		}
+
+		all, err := service.Images(ctx, imagetypes.ListOptions{Manifests: true, SharedSize: true})
+		assert.NilError(t, err)
+
+		assert.Assert(t, is.Len(all, 1))
+		assert.Check(t, is.Equal(all[0].Size, allTotalSize-indexSize-arm64LayerSize-amd64LayerSize))
+
+		assert.Assert(t, is.Len(all[0].Manifests, 2))
+		assert.Check(t, is.Equal(all[0].Manifests[0].Size.Content, arm64ManifestSize+arm64ConfigSize))
+		assert.Check(t, is.Equal(all[0].Manifests[1].Size.Content, amd64ManifestSize+amd64ConfigSize))
+
```

Comment:
Stray empty line at end of body (I know some linter probably will start to complain)